File: C/Recursion/program3.c
//Accept the number from user and display the addition of that number from 1 to that number.

// Input : 4
// Output : 10  (1+2+3+4)

// Input : 5
// Output : 15  (1+2+3+4+5)


#include <stdio.h>

int Summation(int iNo)
{
    static int iSum = 0;
    static int iCnt = 1;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    if(iCnt <= iNo)
    {
        iSum = iSum + iCnt;
        iCnt++;
        Summation(iNo);
    }
    return iSum;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter number\n");
    scanf("%d", &iValue);

    iRet = Summation(iValue);
    printf("Summation is : %d\n", iRet);

    return 0;
}

File: Java/Digit/program3.java
//Accept number from user and display its digits

// Input : 7856
// Output :
// 6
// 5
// 8
// 7

import java.util.*;

class Digit
{
    public int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number");
        iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("The number is: "+this.iNo);
    }

    public void DisplayDigits()
    {
        int iDigit = 0;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        while(iNo != 0)
        {
            iDigit = iNo % 10;
            System.out.println(iDigit);
            iNo = iNo / 10;
        }
    }
}

class program3
{
    public static void main(String arg[])
    {
        Digit obj = new Digit();

        obj.Accept();
        obj.Display();

        obj.DisplayDigits();
    }
}